### PR TITLE
Add method to regenerate PDFs for specified units

### PR DESCRIPTION
**Pull request by bethanyaconnor**
This PR does three things:
1. (the biggest one) adds a method to regenerate PDFs for specified units. We used to have something that could do this but I think it was lost in a refactor somewhere. In cases where the PDF is corrupted in some way, I find it very useful to have a method to do this. I did pretty much copy and adjust the code from `generate_missing_pdfs` -- I can't see a good way to reduce code duplication here but open to suggestions!
2. Commented out resource PDF generation in both `generate_missing_pdfs` and `regenerate_pdfs`. This was broken by the Ruby 3 upgrade and is tracked [here](https://codedotorg.atlassian.net/browse/TEACH-601).
3. Fixed a minor bug where we were inserting a page in the script overview PDF when there were lessons without lesson plans. You can see this by going to https://studio.code.org/s/csd3-2022, printing lesson plans, and scrolling to the bottom of the PDF. Lesson 1 is inserted as the second to last lesson in the place of a lesson plan-less lesson.

### CODE REVIEW

**Comment by rshipp** [comment]
does this mean we can revert #52774 and get partial regeneration (minus the resource pdfs) back, or is that still broken?

File: dashboard/lib/services/curriculum_pdfs.rb [hunk 1]
```diff
@@ -121,6 +121,39 @@ def self.get_pdfless_lessons(script)
       end
     end
 
+    def self.regenerate_pdfs(scripts)
```
Comment (rshipp) [inline dashboard/lib/services/curriculum_pdfs.rb:124]: just checking my understanding: this is just for manual use from a console?

**Review by rshipp — APPROVED** [review]

File: dashboard/lib/services/curriculum_pdfs.rb (hunk 1 above)
Comment (bethanyaconnor) [reply dashboard/lib/services/curriculum_pdfs.rb:124]: Yep!

**Comment by bethanyaconnor** [comment]
> does this mean we can revert https://github.com/code-dot-org/code-dot-org/pull/52774 and get partial regeneration (minus the resource pdfs) back, or is that still broken?

Yep! I was going to do that as a separate PR so that I can monitor the first staging build that contains it, just in case it breaks the staging build again

**Comment by bethanyaconnor** [comment]
Copying over some addition context from https://codedotorg.atlassian.net/browse/TEACH-620:

I’m going to do 3 things:

1. Add a method to regenerate PDFs for specified units in cases where the PDFs are corrupted
2. Regenerate the PDFs that we know are in a bad state
3. Re-enable overview PDF generation on staging

**Pull request merged**